ui/src/main/java/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -2795,16 +2800,35 @@ public void delConnection() {
   }
 
   public void sqlConnection() {
-    final DatabaseMeta databaseMeta = (DatabaseMeta) selectionObject;
+    SpoonTreeLeveledSelection leveledSelection = (SpoonTreeLeveledSelection) selectionObject;
+    DatabaseManagementInterface dbManager = null;
+    DatabaseMeta databaseMeta = null;
+    try {
+      dbManager = getDbManager( leveledSelection.getLevel() );
+      databaseMeta = dbManager.getDatabase( leveledSelection.getName() );
+    } catch ( Exception e ) {
+      new ErrorDialog( shell, BaseMessages.getString( PKG, "Spoon.ErrorDialog.Title" ),
+        BaseMessages.getString( PKG, "Unexpected error retrieving Database Connection", databaseMeta.getName() ), e );
+    }
     delegates.db.sqlConnection( databaseMeta );
```

Comment:
All of these calls to delegates.db should be inside the try block now right?  Some of them are and some aren't.

Reply:
I reused the existing code and the SpoonDbDelegate code is not throwing any exception.

Reply:
Right, but if there was an exception in the try block, and the dialog was displayed, we probably do not want to proceed with the call to the delegate.

Reply:
And I see that you copied this from editConnection(), which I messed up in the first place. ;)

Reply:
I'm actually surprised the compiler isn't complaining that databaseMeta might not have been initialized.